spec/features/user_returns_checked_out_book_spec.rb

```diff
@@ -0,0 +1,24 @@
+require "rails_helper"
+
+feature "User returns only books she borrowed from the library" do
+  scenario "successfully" do
+    book_1 = create(:book)
+    patron_1 = create(:patron)
+    sign_in_and_check_out_book(patron_1)
```

Comment:
def `sign_in_and_check_out_book` can probably also handle book and patron creation, except I wanted to make it clear that book creation is an not completed by the user. Additionally, creating a patron inside the function made it very difficult to tell that after sign-out a different user is signing in. Any advice you have on how to keep the functions small, but also keep the clear and readable would be great.